clients/drcachesim/tests/window_test.cpp

```diff
@@ -0,0 +1,185 @@
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL GOOGLE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/* Test for multi-window traces where it inserts sleep between windows and
+ * checks that the timestamp for second buffer in a new window is closer to the
+ * previous timestamp than the sleep time.
+ */
+
+#include <assert.h>
+#include <chrono>
+#include <iostream>
+#include <string.h>
+#include <thread>
+#include "analysis_tool.h"
+#include "dr_api.h"
+#include "scheduler.h"
+#include "tracer/raw2trace.h"
+#include "tracer/raw2trace_directory.h"
+
+namespace dynamorio {
+namespace drmemtrace {
+
+namespace {
+
+constexpr int SECONDS_TO_SLEEP = 4;
+constexpr int SECONDS_TO_TIMESTAMP = 1000000;
+
+} // namespace
+
+bool
+my_setenv(const char *var, const char *value)
+{
+#ifdef UNIX
+    return setenv(var, value, 1 /*override*/) == 0;
+#else
+    return SetEnvironmentVariable(var, value) == TRUE;
+#endif
+}
+
+int
+fib_with_sleep(int n)
+{
+    dr_fprintf(STDERR, "Calculating fibonacci of: %d\n", n);
+    std::this_thread::sleep_for(std::chrono::seconds(SECONDS_TO_SLEEP));
+    if (n <= 1)
+        return 1;
+    return fib_with_sleep(n - 1) + fib_with_sleep(n - 2);
+}
+
+static std::string
+post_process()
+{
+    const char *raw_dir;
+    drmemtrace_status_t mem_res = drmemtrace_get_output_path(&raw_dir);
+    assert(mem_res == DRMEMTRACE_SUCCESS);
+    std::string outdir = std::string(raw_dir) + DIRSEP + "trace";
+    void *dr_context = dr_standalone_init();
+    {
```

Comment:
Why is this new scope needed?  Not obvious.  Either add a comment or remove the scope.

Reply:
It was an artifact from another test this one was based on. Removed it.